src/Sentry/SentryLog.cs

```diff
@@ -0,0 +1,281 @@
+        {
+            value = (long)attribute.Value;
+            return true;
+        }
+
+        value = 0L;
+        return false;
+    }
+
+    /// <summary>
+    /// Gets the attribute value associated with the specified key when of type <see cref="double"/>.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    /// <remarks>
+    /// Returns <see langword="true"/> if the <see cref="SentryLog"/> contains an attribute with the specified key of type <see cref="double"/>.
+    /// Otherwise <see langword="false"/>.
+    /// </remarks>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public bool TryGetAttribute(string key, out double value)
+    {
+        if (_attributes.TryGetValue(key, out var attribute) && attribute.Type == "double")
+        {
+            value = (double)attribute.Value;
+            return true;
+        }
+
+        value = 0.0;
+        return false;
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="string"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, string value)
+    {
+        _attributes[key] = new SentryAttribute(value, "string");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="bool"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, bool value)
+    {
+        _attributes[key] = new SentryAttribute(value, "boolean");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="long"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, long value)
+    {
+        _attributes[key] = new SentryAttribute(value, "integer");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="double"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, double value)
+    {
+        _attributes[key] = new SentryAttribute(value, "double");
+    }
+
+    internal void SetAttributes(SentryOptions options)
+    {
+        var environment = options.SettingLocator.GetEnvironment();
+        SetAttribute("sentry.environment", environment);
+
+        var release = options.SettingLocator.GetRelease();
+        if (release is not null)
+        {
+            SetAttribute("sentry.release", release);
+        }
+
+        SetAttribute("sentry.sdk.name", Constants.SdkName);
```

Comment:
we modify the SDK name when using the SDK from a 'wrapping' SDK such as `Sentry.Maui` and `Sentry.AspNetCore`. So we'll want to have a place where we know where to pull the right SDK name from.

Now, it'll just report as `sentry.dotnet` even if the end user installed `Sentry.Maui` in their app